libp2p-swarm/src/connection_reuse.rs

```diff
@@ -159,27 +164,32 @@ where
 /// Implementation of `Stream<Item = (impl AsyncRead + AsyncWrite, Multiaddr)` for the
 /// `ConnectionReuse` struct.
 pub struct ConnectionReuseListener<S, M>
-where
-	S: Stream<Item = (M, Multiaddr), Error = IoError>,
-	M: StreamMuxer,
+where S: Stream<Item = (Result<M, IoError>, Multiaddr), Error = IoError>,
```

Comment:
Can you keep the old style here (`where` on its own line)? It makes the changes more obvious and it's what `rustfmt` will change the style to anyway when we finally get that run over the codebase.